frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,116 @@
+import { styled } from "solid-styled-components";
+import { createFormGroup, createFormControl } from "solid-forms";
+import Button from "../components/form/button";
+import EmailInput from "../components/form/email-input";
+import HashInput from "../components/form/hash-input";
+import FileInput from "../components/form/file-input";
+import Slider from "../components/slider";
+import { getFilesUrls, validateEmail } from "../helpers";
+import { postTrain } from "../api";
+import { config } from "../config";
+
+const Train = () => {
+  const group = createFormGroup({
+    email: createFormControl("", {
+      validators: (value) =>
+        validateEmail(value) ? null : { errorMessage: "Invalid email address" },
+    }),
+    files: createFormControl([], {
+      validators: (files) =>
+        files.length < config.minFiles
+          ? { errorMessage: `Number files must be ${config.minFiles} or more` }
+          : null,
+    }),
+    hash: createFormControl("", { readonly: true, disabled: true }),
+  });
+
+  const imageList = () => getFilesUrls(group.controls.files.value);
+
+  const handleFormSubmit = async () => {
+    if (group.isSubmitted) {
+      return;
+    }
+    if (!group.controls.email.isValid) {
+      group.controls.email.markTouched(true);
+    } else if (!group.controls.files.isValid) {
+      group.controls.files.markTouched(true);
+    } else {
+      // temp const
+      const isMock = true;
+
+      group.markPending(true);
+      const hash = await postTrain(
+        group.controls.email.value,
+        group.controls.files.value,
+        isMock
+      );
+      group.controls.hash.setValue(hash);
+      group.controls.hash.markDisabled(false);
+      group.markSubmitted(true);
```

Comment:
Please disable the `Train` button when submitted and enable when the `imageList` changes